packages/local-resolver/src/parsePref.ts

```diff
@@ -57,7 +62,7 @@ function fromLocal (
     .replace(/^(file|link|workspace):[/]*([A-Za-z]:)/, '$2') // drive name paths on windows
     .replace(/^(file|link|workspace):(?:[/]*([~./]))?/, '$2')
 
-  const protocol = type === 'directory' ? 'link:' : 'file:'
+  const protocol = type === 'directory' ? (injected ? 'local/' : 'link:') : 'file:'
```

Comment:
why we use 'local/' instead of something like 'local:', 'local/' does not look like a protocol, maybe something like '_pnpm_local:' would be more semantic

Reply:
it is just a convention that we use. The part before the first slash is the source of the package. So a git-hosted package would be `github.com/foo/bar`

Reply:
I will change it to `file:`. We use `file:` for local tarball dependencies now.

Reply:
updated